clients/src/main/java/org/apache/kafka/common/record/AbstractLegacyRecordBatch.java

```diff
@@ -264,26 +263,48 @@ static void writeHeader(DataOutputStream out, long offset, int size) throws IOEx
     }
 
     private static final class DataLogInputStream implements LogInputStream<AbstractLegacyRecordBatch> {
-        private final DataInputStream stream;
+
+        private static final int OFFSET_OFFSET = 0;
+        private static final int SIZE_OFFSET = 8;
+        private static final int BUFFER_SIZE = 8 // offset
+                                               + 4; // size
```

Comment:
Yeah, my PR removes all of these in favour of existing constants: https://github.com/apache/kafka/pull/3164/commits/a7bc3e6e98ad3f983f5619601b216e83142afe42